File: c++/10816.cpp
// my answer
// #include <iostream>
// #include <vector>

// using namespace std;

// int main(){
//   ios_base::sync_with_stdio(false);
//   cin.tie(NULL);
//   cout.tie(NULL);

//   int n;
//   vector<int> nCard(20000000);
//   cin >> n;
//   for(int i=0; i<n; i++){
//     int a;
//     cin >> a;
//     nCard[a+10000000]++;
//   }
//   int m;
//   cin >> m;
//   for(int i=0; i<m; i++){
//     int b;
//     cin >> b;
//     cout << nCard[b+10000000] << ' ';
//   }

//   return 0;
// }

// another answer
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(){
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  int n;
  cin >> n;

  vector<int> v(n);
  for(int i=0; i<n; i++){
    cin >> v[i];
  }
  sort(v.begin(), v.end());

  int m;
  cin >> m;
  for(int i=0; i<m; i++){
    int a;
    cin >> a;
    auto l = lower_bound(v.begin(), v.end(), a);
    auto u = upper_bound(v.begin(), v.end(), a);
    cout << u-l << ' ';
  }
}

File: js/10816.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const arr = inputs[1]
  .split(' ')
  .map(Number)
  .sort((a, b) => a - b);

const answer = [];

inputs[3].split(' ').forEach((e) => {
  const ub = upperBound(+e);
  const lb = lowerBound(+e);
  answer.push(ub - lb);
});

console.log(answer.join('\n'));

function lowerBound(target) {
  let start = 0;
  let end = arr.length;

  while (start < end) {
    const mid = Math.floor((start + end) / 2);
    if (arr[mid] >= target) {
      end = mid;
    } else {
      start = mid + 1;
    }
  }

  return end;
}

function upperBound(target) {
  let start = 0;
  let end = arr.length;

  while (start < end) {
    const mid = Math.floor((start + end) / 2);
    if (arr[mid] > target) {
      end = mid;
    } else {
      start = mid + 1;
    }
  }

  return end;
}
